exercises/danny_hindson/ex02_oo_basics/src/ShapeSorter.h

```diff
@@ -0,0 +1,26 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#ifndef SHAPESORTER_H_
+#define SHAPESORTER_H_
+
+#include "Shape.h"
+#include <vector>
+
+class ShapeSorter {
+public:
+  ShapeSorter(std::vector<Shape*> v) : shapeVec(v) {};
```

Comment:
This is perfectly valid code but causes a copy of the vector that can be avoided using `std::move` from C++11.
```c++
ShapeSorter(std::vector<Shape*> v) : shapeVec(std::move(v)) {};
```
I'd suggest a quick read over something about [move semantics](https://mbevin.wordpress.com/2012/11/20/move-semantics/)

Reply:
OK thanks. Think I've got that. That was new to me